boards/arm/twr_ke18f/pinmux.c

```diff
@@ -43,35 +44,121 @@ static int twr_ke18f_pinmux_init(const struct device *dev)
 	pinmux_pin_set(porte, 11, PORT_PCR_MUX(kPORT_MuxAlt2));
 #endif
 
-#if DT_NODE_HAS_STATUS(DT_NODELABEL(lpspi0), okay) && CONFIG_SPI
-	/* SPI0 SCK, SIN, SOUT */
-	pinmux_pin_set(porte, 0, PORT_PCR_MUX(kPORT_MuxAlt2));
-	pinmux_pin_set(porte, 1, PORT_PCR_MUX(kPORT_MuxAlt2));
-	pinmux_pin_set(porte, 2, PORT_PCR_MUX(kPORT_MuxAlt2));
-#endif
-#ifdef CONFIG_BOARD_TWR_KE18F_SPI_0_PCS2
-	/* SPI0 PCS2 */
-	pinmux_pin_set(porte, 6, PORT_PCR_MUX(kPORT_MuxAlt2));
+#if CONFIG_BOARD_TWR_KE18F_SPI_0_PCS2 ||	\
+	CONFIG_BOARD_TWR_KE18F_SPI_1_PCS0 ||	\
+	CONFIG_BOARD_TWR_KE18F_SPI_1_PCS2 ||	\
+	CONFIG_BOARD_TWR_KE18F_FLEXIO_CLKOUT
+	int err; /* Used by pinctrl functions */
 #endif
 
-#if DT_NODE_HAS_STATUS(DT_NODELABEL(lpspi1), okay) && CONFIG_SPI
-	/* SPI1 SCK, SIN, SOUT */
-	pinmux_pin_set(portd, 0, PORT_PCR_MUX(kPORT_MuxAlt3));
-	pinmux_pin_set(portd, 1, PORT_PCR_MUX(kPORT_MuxAlt3));
-	pinmux_pin_set(portd, 2, PORT_PCR_MUX(kPORT_MuxAlt3));
-#endif
-#ifdef CONFIG_BOARD_TWR_KE18F_SPI_1_PCS0
-	/* SPI1 PCS0 */
-	pinmux_pin_set(portd, 3, PORT_PCR_MUX(kPORT_MuxAlt3));
+	/* Configuation for SPI0 PCS2 KConfig */
+#if CONFIG_BOARD_TWR_KE18F_SPI_0_PCS2 && CONFIG_SPI
+	/* Struct is declared as extern here, LPSPI driver implements it */
+	extern struct pinctrl_dev_config
+		Z_PINCTRL_DEV_CONFIG_NAME(DT_NODELABEL(lpspi0));
```

Comment:
please use `PINCTRL_DT_DEV_CONFIG_DECLARE`. See `samples/boards/nrf/dynamic_pinctrl` on how to use dynamic pinctrl.

Reply:
Fixed, thank you for the pointer